crates/ruma-common/src/events/room/power_levels.rs

```diff
@@ -265,6 +238,18 @@ pub struct RoomPowerLevels {
     pub notifications: NotificationPowerLevels,
 }
 
+impl RoomPowerLevels {
```

Comment:
I don't know if it makes sense in Ruma, but in fractal we also have a method to check if a user can do a certain action. It could take an enum as parameter:

```rust
pub enum PowerLevelAction {
  Ban,
  Invite,
  Kick,
  Redact,
  Event(RoomEventType),
}

/// Whether the given user can do the given action with these power levels.
pub fn can_user_send(&self, user_id: &UserId, action: PowerLevelAction) -> bool {
  // …
}
```